src/models/graph.ts

```diff
@@ -474,7 +497,7 @@ export class Graph<N extends INodeBase, E extends IEdgeBase> implements IGraph<N
     this._edges.removeMany(removedEdgeIds);
   }
 
-  private _removeNodes(nodeIds: any[]) {
+  private _removeNodes(nodeIds: any[]): { nodeIds: any[]; edgeIds: any[] } {
```

Comment:
This return type is cool. I would maybe extract it into an interface/type. Also, just for a sake of consistency, I would maybe consider returning the same type on `_removeEdges` - right now, nodeIds will be always empty, but in the future if we add "removeConnectingNodes` for edges, then the API still holds and can use the returned object with both `nodeIds` and `edgeIds`.